### DISCUSSION TITLE
Simply add special font using @font-face in custom CSS

### DISCUSSION THREAD

**Discussion by Vinc26** [GetPublii/Publii#1750] [Help]
I'm trying to do a so simple thing. I want to have only a small part of my text using a different font.

Yes, I've read https://getpublii.com/docs/fonts.html : but I don't want to apply the special font for all my website. Only for a small text.

So I'm using the well known @font-face in my custom CSS:

```
@font-face{
  font-family: 'Caveat';
  src: url('/Caveat-VariableFont_wght.ttf');
}

.manu {
  font-family: "Caveat", cursive;
  font-optical-sizing: auto;
  font-weight: 400;
  font-style: normal;
}
```

and in my page: `<span class='manu'>small text</span>`

I've also uploaded the font `Caveat-VariableFont_wght.ttf` using the file manager of Publii.

But, instead of using Caveat, it uses the default cursive font. Why?

You can try it here: [https://aurelielamour.com/fonttest/](https://aurelielamour.com/fonttest/)

You can also see that it's not the good font: https://fonts.google.com/specimen/Caveat

**Comment by Vinc26** [comment]
For now, the only solution I founded is to use:
```
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
<link href="https://fonts.googleapis.com/css2?family=Caveat:wght@400..700&display=swap" rel="stylesheet">
```

But I don't like to connect to Google. It is not good for privacy and for RGDP in Europe.

Please help me to better understand how to add and use specials fonts inside en text; not for all my website.

**Comment by candidexmedia** [comment]
> I've also uploaded the font Caveat-VariableFont_wght.ttf using the file manager of Publii.

The URL in your `font face` source wasn't pointing to your actual file — in fact, it was pointing to nothing, so it fell back to the next font listed: "cursive". The `link rel` works because you're pointing to a file hosted on another server. It's the same issue as your other [discussion post about the JavaScript file](https://github.com/GetPublii/Publii/discussions/1751).

> Please help me to better understand how to add and use specials fonts inside en text; not for all my website.

If you want to self-host your font files, add them to your site's theme folder directly (not the GUI) and edit the `main.css` file in the theme folder. Check out my gist for more info: https://gist.github.com/candidexmedia/fc7d21c470631ddbbffa4bd7fae14d0f

The docs are also a good resource: https://getpublii.com/docs/fonts.html#adding-custom-font

**Reply by Vinc26** [reply]
Thanks :) If I understand correctly, there is a problem with the method provided by Publii (adding-custom-font): It is only to modify the entire font of the site. There, I am simply trying to change the font of a text.

I see that I will be forced to modify the theme with an override. But the day the theme will be updated by Publii, I will have to resume all my changes :(. That's why I preferred to use only Custom CSS and Custom HTML.

Unless I misunderstood something?

**Reply by candidexmedia** [reply]
> I see that I will be forced to modify the theme with an override. But the day the theme will be updated by Publii, I will have to resume all my changes :(.

The override folder will be preserved during theme upgrades. Thst said, you may need to git.diff any duplicated files (like `main.css`) from your override against the same file in the new theme, to make sure it's updated properly.

> That's why I preferred to use only Custom CSS and Custom HTML.

You have to make a decision about which is more important to you: the simplicity of Custom CSS + HTML, or being able to host the files yourself?

**Comment by michaelfluegge** [comment]
I would start with changing the path from relative   src: url('/Caveat-VariableFont_wght.ttf');
to absolute, like. src: url('https://yoururl/assets/font/Caveat-VariableFont_wght.ttf');  or whatever the path is. 
If that works, I would try to change it to the correct relative path. You may find information about relative paths here for example: https://css-tricks.com/quick-reminder-about-file-paths/

**Reply by Vinc26** [reply]
Thank you, yes, the solution would be to use absolute urls. But it doesn't work locally.
